packages/@react-spectrum/tag/src/TagGroup.tsx

```diff
@@ -70,33 +73,116 @@ function TagGroup<T extends object>(props: SpectrumTagGroupProps<T>, ref: DOMRef
   }, state, domRef);
   const {tagGroupProps} = useTagGroup(props);
 
+  let [isCollapsed, setIsCollapsed] = useState(maxRows != null);
+  let [tagState, setTagState] = useValueEffect({visibleTagCount: gridCollection.size, showCollapseButton: false});
+
+  let updateVisibleTagCount = useCallback(() => {
+    if (maxRows > 0) {
+      let computeVisibleTagCount = () => {
+        // Refs can be null at runtime.
+        let currDomRef: HTMLDivElement | null = domRef.current;
+        if (!currDomRef) {
+          return;
+        }
+
+        let [tags, button] = [...currDomRef.children].reduce((acc, el) => {
+          if (el.tagName !== 'BUTTON') {
+            acc[0].push(el);
+          } else {
+            acc[1] = el;
+          }
+          return acc;
+        }, [[], null]);
+        let currY = -Infinity;
+        let rowCount = 0;
+        let index = 0;
+        let tagWidths = [];
+        // Count rows and show tags until we hit the maxRows.
+        for (let tag of tags) {
+          let {y} = tag.getBoundingClientRect();
```

Comment:
```suggestion
          let {width, y} = tag.getBoundingClientRect();
```
then you can delete the getBoundingClientRect call below on 112